executor/hash_table.go

```diff
@@ -43,12 +45,60 @@ const (
 	estCountDivisor = 8
 )
 
+type bitmap struct {
+	bits []byte
+}
+
+func (bm *bitmap) clear() {
+	for i := 0; i < len(bm.bits); i++ {
+		bm.bits[i] = byte(0)
+	}
+}
+
+func (bm *bitmap) reserve(size int) {
+	size = (size + 7) / 8
+	if len(bm.bits) < size {
+		bm.bits = make([]byte, size)
+	} else {
+		bm.clear()
+	}
+}
+
+func (bm *bitmap) isSet(idx int) bool {
+	b := bm.bits[idx/8] & (1 << uint(idx%8))
```

Comment:
```suggestion
	b := bm.bits[idx>>3] & (1 << uint(idx&7))
```

Reply:
The `bitmap` has been replaced with `[]bool`, as recommended by @SunRunAway